### DISCUSSION TITLE
Weight units

### DISCUSSION THREAD

**Discussion by johnp3004** [inventree/InvenTree#6588] [Q&A]
Hi, first off, completely new user of Inventree - but it looks very promising for our needs!
Also, sorry if this is a "dumb" question.

I am trying to use InvenTree as a inventory and production system for our brewery, and I am trying to set up products and ingredients (both of them as parts), and then creating a bill of materials for a product containing the different ingredients we use for each product/bottle of brew.

I have added a ingredient and entered kg as units of measure for that ingredient.

When I add a part(ingredient) to the BOM of that product, I want to use grams for the amount of ingredient used for that product (i want to track the total quantity of the ingredient in kg, and for "recipes" / Bill of materials I want to use grams) I get an error when entering 0.75 in the quantity: Quantity must be integer value for trackable parts.

It only allows me to enter numbers in the quantity field so I can't attach a g at the end to define grams.
I tried setting up parts parameters, but can't seem to get this right either.

I feel like I am just having a large brain-fart - but the docs are not really clear on how conversions are handled or defined, and the instructions for physical units in settings are even more vague as it just links to Pints docs, when actually changing the settings for physical units it doesnt really explain what to enter in what text-box.

If anyone could explain how I can achieve what I wish, or give me a more detailed explanation as to how unit conversions work/can be defined in InvenTree it would be fantastic!

**Comment by SchrodingersGat** [comment]
> When I add a part(ingredient) to the BOM of that product, I want to use grams for the amount of ingredient used for that product (i want to track the total quantity of the ingredient in kg, and for "recipes" / Bill of materials I want to use grams)

This feature is not yet supported - but you can [sponsor this active issue](https://github.com/inventree/InvenTree/issues/4922) to give it some priority.

Currently, BOM items must be specified in the same units as the linked part. i.e. in this case, `kg`.

> If anyone could explain how I can achieve what I wish, or give me a more detailed explanation as to how unit conversions work/can be defined in InvenTree it would be fantastic!

A good starting point would be the docs - if there are any areas there where you feel there are some gaps in the documentation, please raise an issue - or you could contribute to the documentation too :)

- https://docs.inventree.org/en/latest/concepts/units/
- https://docs.inventree.org/en/latest/part/part/#units-of-measure

> I get an error when entering 0.75 in the quantity: Quantity must be integer value for trackable parts.

This issue is due to the fact that you have marked the part in question as `trackable` - https://docs.inventree.org/en/latest/part/trackable/

If you remove this requirement, you should be able to add it to the BOM in a non-integer quantity
